cvat-core/src/annotations-objects.js

```diff
@@ -248,6 +248,19 @@
             this.label = label;
             this.attributes = {};
             this.appendDefaultAttributes(label);
+
+            // Try to keep old attributes if name and inputType matches and old value is still valid
+            for (const attribute of redoLabel.attributes) {
+                for (const oldAttribute of undoLabel.attributes) {
+                    if (
+                        attribute.name === oldAttribute.name
+                        && attribute.inputType === oldAttribute.inputType
```

Comment:
input types "select" and "radio" are fully compatible
we can also check compatibility additionally between "text" and "radio' or "select"